src/lib/array/removeAt.ts

```diff
@@ -0,0 +1,7 @@
+export const removeAt = <T = any>(array: T[], idx: number): T[] => {
+  if (idx >= array.length) {
+    throw new Error(`removeAt: Index ${idx} out of bounds`)
+  }
+
+  return [...array.slice(0, idx), ...array.slice(idx + 1, array.length)]
```

Comment:
a small codegolf comment but... `...array.slice(0, idx).concat(array.slice(idx + 1))` would also do the job